src/main/java/Jimmy.java

```diff
@@ -0,0 +1,82 @@
+import essentials.Parser;
+import essentials.Storage;
+import essentials.TaskList;
+import essentials.Ui;
+import exceptions.JimmyException;
+
+import java.util.Scanner;
+
+/**
+ * Represents the main bot class.
+ */
+public class Jimmy {
+    private final Storage storage;
+    private TaskList tasks;
+    private final Ui ui;
+    private final Parser parser;
+
+    /**
+     * Constructor for Jimmy bot.
+     *
+     * @param filePath The file path to the file where the tasks are stored.
+     */
+    public Jimmy(String filePath) {
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage);
+        } catch (JimmyException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Run the bot.
+     */
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+        ui.greetUser();
+        boolean isBotRunning = true; // flag to check if bot is running
+
+        try {
+            try {
+                while (isBotRunning) {
+                    String userInput = sc.nextLine();
+                    String[] parsedInput = parser.parseUserInput(userInput);
+                    String instruction = parsedInput[0];
+                    String details = parsedInput[1];
+                    switch (instruction) {
+                    case "bye":
```

Comment:
I see that your indentation when using switch is correct and follows the coding style guide. I find that this is a common mistake that a lot of people make, so its good that you avoided this mistake!